### ISSUE TITLE
`systemd-networkd-wait-online` with `networking.useNetworkd` broken as of systemd 253.6

### ISSUE THREAD

**Issue by winterqt** [NixOS/nixpkgs#247608]
https://github.com/NixOS/nixpkgs/blob/48fecd04896121f8af12ff33754ef251d4ae6318/nixos/modules/tasks/network-interfaces-systemd.nix#L62-L70

This configuration has worked fine until https://github.com/systemd/systemd/commit/2f96a29c2c55bdd67cdd8e0b0cfd6971968e4bca, which now makes it so wait-online will indefinitely hang if no managed interfaces are configured that have `RequiredForOnline` set, which our default configuration for `networking.useNetworkd` does not.

@NixOS/systemd (specifically @ElvishJerricco, as the original author of the above configuration), how do we want to handle this?

**Comment by arianvp** [comment]
I can reproduce

**Comment by Mic92** [comment]
Yes. It's broken and we also disables this in srvos everywhere: https://github.com/numtide/srvos/blob/85fed23fce8ada8017486090569c0a9054875d5c/nixos/common/networking.nix#L17

**Comment by Majiir** [comment]
I ran into this with `systemd.network.enable` _without_ `networking.useNetworkd`. I had no interfaces configured with networkd. While this isn't a particularly useful configuration, :slightly_smiling_face: it shouldn't cause system activation to hang.

**Comment by ElvishJerricco** [comment]
This will be fixed by #258680, no?

**Comment by misuzu** [comment]
I've just run into this on the [latest](https://github.com/NixOS/nixpkgs/commit/b8697e57f10292a6165a20f03d2f42920dfaf973) nixos-unstable, this combination triggers the issue:
```
networking.useNetworkd = true
networking.networkmanager.enable = true;
```

`systemd.network.wait-online.anyInterface = true;` doesn't help

**Comment by akvadrako** [comment]
Related issue: https://github.com/systemd/systemd/issues/25813

**Comment by asymmetric** [comment]
So is the solution to default to `systemd.network.wait-online.enable = false;` if `networking.useNetworkd = true`?

Also, not sure if related, but the documentation for `boot.initrd.systemd.network.wait-online.enable` says

> systemd-networkd-wait-online can timeout and fail if there are no network interfaces available for it to manage. When systemd-networkd is enabled but a different service is responsible for managing the system’s internet connection (for example, NetworkManager or connman are used to manage WiFi connections), this service is unnecessary and can be disabled.

So does this mean we should not "wait online" when `networking.useNetworkd && networking.networkmanager.enable`?

**Comment by audishos** [comment]
> So is the solution to default to `systemd.network.wait-online.enable = false;` if `networking.useNetworkd = true`?
> 
> Also, not sure if related, but the documentation for `boot.initrd.systemd.network.wait-online.enable` says
> 
> > systemd-networkd-wait-online can timeout and fail if there are no network interfaces available for it to manage. When systemd-networkd is enabled but a different service is responsible for managing the system’s internet connection (for example, NetworkManager or connman are used to manage WiFi connections), this service is unnecessary and can be disabled.
> 
> So does this mean we should not "wait online" when `networking.useNetworkd && networking.networkmanager.enable`?

I mkForce disabled wait-online and that seems to have fixed it.
```nix
  systemd.network.wait-online.enable = lib.mkForce false;
```
